app/lib/features/chat/widgets/chats_list.dart

```diff
@@ -15,95 +15,100 @@ import 'package:logging/logging.dart';
-              );
-            },
-            loading: () => const SliverToBoxAdapter(
-              child: Center(
-                heightFactor: 10,
-                child: CircularProgressIndicator(),
-              ),
-            ),
-            error: (e, s) {
-              _log.severe('Failed to filter convos', e, s);
-              return SliverToBoxAdapter(
-                child: Center(
-                  heightFactor: 10,
-                  child: Text(L10n.of(context).searchingFailed(e)),
-                ),
-              );
-            },
-            skipLoadingOnReload: true,
-          );
+  Widget build(BuildContext context, WidgetRef ref) {
+    if (ref.watch(hasRoomFilters)) {
+      return _renderFiltered(context, ref);
     }
     final chats = ref.watch(chatIdsProvider);
 
     if (chats.isEmpty) {
-      final hasFirstSynced =
-          ref.watch(syncStateProvider.select((x) => !x.initialSync));
-      if (!hasFirstSynced) {
+      if (!ref.watch(hasFirstSyncedProvider)) {
+        return _renderSyncing(context);
+      }
+      return _renderEmpty(context);
+    }
+    return _renderList(context, chats);
+  }
+
+  Widget _renderFiltered(BuildContext context, WidgetRef ref) {
+    final filteredChats = ref.watch(filteredChatsProvider);
+    return filteredChats.when(
+      data: (chatsIds) {
+        if (chatsIds.isEmpty) {
+          return SliverToBoxAdapter(
+            child: Center(
+              heightFactor: 10,
+              child: Text(L10n.of(context).noChatsFoundMatchingYourFilter),
+            ),
+          );
+        }
+        return _renderList(
+          context,
+          chatsIds,
+        );
+      },
+      loading: () => const SliverToBoxAdapter(
+        child: Center(
+          heightFactor: 10,
+          child: CircularProgressIndicator(),
+        ),
+      ),
+      error: (e, s) {
+        _log.severe('Failed to filter convos', e, s);
         return SliverToBoxAdapter(
           child: Center(
-            heightFactor: 1.5,
-            child: EmptyState(
-              title: L10n.of(context).noChatsStillSyncing,
-              subtitle: L10n.of(context).noChatsStillSyncingSubtitle,
-              image: 'assets/images/empty_chat.svg',
-            ),
+            heightFactor: 10,
+            child: Text(L10n.of(context).searchingFailed(e)),
           ),
         );
-      }
-      return SliverToBoxAdapter(
-        child: Center(
-          heightFactor: 1.5,
-          child: EmptyState(
-            title: L10n.of(context).youHaveNoDMsAtTheMoment,
-            subtitle: L10n.of(context).getInTouchWithOtherChangeMakers,
-            image: 'assets/images/empty_chat.svg',
-            primaryButton: ActerPrimaryActionButton(
-              onPressed: () async => context.pushNamed(
-                Routes.createChat.name,
-              ),
-              child: Text(L10n.of(context).sendDM),
+      },
+      skipLoadingOnReload: true,
+      skipLoadingOnRefresh: true,
```

Comment:
the only actual change was adding this line to avoid switching to the loading animation even if for just one frame...